docs/csharp/csharp-6.md

```diff
@@ -1,17 +1,570 @@
+only when a debugger is not attached:
+
+[!code-csharp[LogExceptionDebugger](../../samples/snippets/csharp/new-in-6/program.cs#LogExceptionDebugger)]
+
+After adding this in code, you set your debugger to break on all unhandled
+exceptions. Run the program under the debugger, and the debugger breaks
+whenever `PerformFailingOperation()` throws a `RecoverableException`.
+The debugger breaks your program, because the catch clause won't be executed
+due to the false-returning exception filter.
+
+## `nameof` Expressions
+
+The `nameof` expression evaluates to the name of a symbol. It's a great
+way to get tools working with you whenever you need the name of a variable,
+a property, or a member field.
+
+One of the most common uses for `nameof` is to provide the name of a symbol
+that caused an exception:
+
+[!code-csharp[nameof](../../samples/snippets/csharp/new-in-6/NewCode.cs#UsingStaticString)]
+
+Another use is with XAML based applications that implement the `INotifyPropertyChanged`
+interface:
+
+[!code-csharp[nameofNotify](../../samples/snippets/csharp/new-in-6/viewmodel.cs#nameofNotify)]
+
+The advantage of using the `nameof` operator over a constant string is
+that tools can understand the symbol. If you use refactoring tools to
+rename the symbol, it will rename it in the `nameof` expression. Constant
+strings don't have that advantage. Try it yourself in your favorite editor:
+Rename a variable, and any nameof expressions will update as well.
+
+The `nameof` expression produces the unqualified name of its argument
+(`LastName` in the examples above) even if you use the fully qualified
+name for the argument:
+
+[!code-csharp[QualifiedNameofNotify](../../samples/snippets/csharp/new-in-6/viewmodel.cs#QualifiedNameofNotify)]
+
+The above `nameof` expression produces `FirstName`, not `UXComponents.ViewModel.FirstName`.
+
+## Await in Catch and Finally blocks
+
+C# 5 had several limitations around where you could place `await` expressions.
+One of those has been removed in C# 6. You can now use `await` in `catch`
+or `finally` expressions. 
+
+The addition of await expressions in catch and finally blocks may appear
+to complicate how those are processed. Let's add an example to discuss
+how this appears. In any async method, you can use an await expression
+in a finally clause.
+
+With C# 6, you can also await in catch expressions. This is most often
+used with logging scenarios:
+
+[!code-csharp[AwaitFinally](../../samples/snippets/csharp/new-in-6/NetworkClient.cs#AwaitFinally)]
+
+The implementation details for adding `await` support inside `catch`
+and `finally` clauses ensures that the behavior is consistent with the
+behavior for synchronous code. When code executed in a `catch` or `finally`
+clause throws, execution looks for a suitable `catch` clause in the next
+surrounding block. If there was a current exception, that exception is
+lost. The same happens with `await`ed expressions in `catch` and `finally`
```

Comment:
`await`ed looks strange in the published output - maybe skip the keyword formatting here?